File: VB/GridViewDetailColumnWidth/MyGridViewColumnMenu1.vb
Imports DevExpress.XtraGrid.Views.Grid
Imports System.Collections.Generic
Imports System
Imports DevExpress.XtraGrid.Menu
Imports DevExpress.Utils.Menu
Imports DevExpress.XtraGrid.Localization

Namespace GridViewDetailColumnWidth
	Public Class MyGridViewColumnMenu
		Inherits GridViewColumnMenu

		Public Sub New(ByVal view As GridView)
			MyBase.New(view)
		End Sub

		Protected Overrides Sub OnMenuItemClick(ByVal sender As Object, ByVal e As EventArgs)
'INSTANT VB NOTE: The variable view was renamed since Visual Basic does not handle local variables named the same as class members well:
			Dim view_Renamed = TryCast(View, MyGridView)
			Dim item = DirectCast(sender, DXMenuItem)
			If Not(TypeOf item.Tag Is GridStringId) OrElse view_Renamed Is Nothing Then
				Return
			End If
			Dim id = DirectCast(item.Tag, GridStringId)
			MyBase.OnMenuItemClick(sender, e)
			If id = GridStringId.MenuColumnRemoveColumn Then
				view_Renamed.AutoSynchronizeDetailsColumnWidths()
			End If
		End Sub
	End Class
End Namespace

File: CS/GridViewDetailColumnWidth/MyGridViewColumnMenu1.cs
using DevExpress.XtraGrid.Views.Grid;
using System.Collections.Generic;
using System;
using DevExpress.XtraGrid.Menu;
using DevExpress.Utils.Menu;
using DevExpress.XtraGrid.Localization;

namespace GridViewDetailColumnWidth {
    public class MyGridViewColumnMenu : GridViewColumnMenu {
        public MyGridViewColumnMenu(GridView view) : base(view) { }

        protected override void OnMenuItemClick(object sender, EventArgs e) {
            var view = View as MyGridView;
            var item = (DXMenuItem)sender;
            if(!(item.Tag is GridStringId) || view == null)
                return;
            var id = (GridStringId)item.Tag;
            base.OnMenuItemClick(sender, e);
            if(id == GridStringId.MenuColumnRemoveColumn)
                view.AutoSynchronizeDetailsColumnWidths();
        }
    }
}